src/inapp/utils.ts

```diff
@@ -114,6 +115,103 @@ export const sortInAppMessages = (messages: Partial<InAppMessage>[] = []) => {
   return messages.sort(by(['priorityLevel', 'asc'], ['createdAt', 'asc']));
 };
 
+/**
+ * detect amount of local storage remaining (quota) and used (usage).
+ * if usageDetails exist (not supported in Safari), use this instead of usage.
+ */
+export const determineRemainingStorageQuota = async () => {
+  try {
+    const storage: BrowserStorageEstimate | undefined =
+      'storage' in navigator && 'estimate' in navigator.storage
+        ? await navigator.storage.estimate()
+        : undefined;
+
+    /** 50 MB is the typical web browser cache quota for mobile devices */
+    const mobileBrowserQuota = 52428800;
+    /** max quota of browser storage that in-apps will potentially fill */
+    const inAppMaxBrowserQuota = storage?.quota && storage.quota * 0.6;
+    /** determine lower max quota that can be used for message cache */
+    const messageQuota =
+      inAppMaxBrowserQuota && inAppMaxBrowserQuota < mobileBrowserQuota
+        ? inAppMaxBrowserQuota
+        : mobileBrowserQuota;
+
+    /** how much local storage is being used */
+    const usage = storage?.usage && storage.usage;
+    const idbUsage =
+      storage?.usageDetails?.indexedDB && storage.usageDetails.indexedDB;
```

Comment:
Do we need the `&&` check if we're optionally chaining?

Reply:
Also since they have similar usage can we combine the assignment into one variable so we don't have to check below which one to use. `usage = storage?.usageDetails?.indexedDB ?? storage?.usage`